### DISCUSSION TITLE
Open source firebase alternative

### DISCUSSION THREAD

**Discussion by rlnorthcutt** [davidmyersdev/octo#230] [Ideas]
I just ran across Octo.app, and it is a very cool app! I love the commitment to open source, security, and transparency. In the spirit of open source, have you considered moving the app (and the default setup) from firebase to Supabase? Supabase is an open source alternative: https://supabase.com/

I'm sure this carries some unknown challenges, but from an OSS perspective it would be neat. Plus, this would enable organizations to host their own Supabase instance to run Octo. I'd want to still see business licensing in this model, but if it worked, then it may open up a new path to business licenses that can help fund growth of the project.

Regardless - great work all around.

**Comment by davidmyersdev** [comment]
I really appreciate you starting this discussion. I've actually been considering Supabase for a little while now for the same reasons you mentioned. The main concern I have is related to migrating OAuth providers (and sessions), but Supabase actually has a decent strategy for that now. Business licensing (with a support model) is something I've been considering recently too, and I'm always open to discussing options there. I'll add both to the roadmap now.

**Comment by rlnorthcutt** [comment]
Once Supabase is available, then a few other options open up. First, there is the possibility of taking the free version and offloading it onto the user's Supabase instance. So, in theory, you could offer the community edition for free with the user setting up their own version. Then, the managed version could carry some small cost to support the project. This would make the project more sustainable without losing the free version. Just an idea...

However, the other VERY interesting idea is to work with Supabase to use Octo as a quick start project similar to this: https://github.com/supabase/examples-archive/tree/main/supabase-js-v1/slack-clone/nextjs-slack-clone

Even just to be listed in the docs would be a great way to bring more attention and users to the project.

**Reply by davidmyersdev** [reply]
I like these ideas. I'm still not quite ready to attack the Supabase migration yet, but I think I'll be ready once I finish the move to Nuxt.

**Comment by raelb** [comment]
Please also consider https://pocketbase.io. Seems newer than Supabase, and not as well known, but has very simple and convenient deployment.

**Reply by davidmyersdev** [reply]
PocketBase looks pretty cool. I'll keep it in mind as an alternative!
